File: 02.Array_problems/InsertionSort.js

// sort the array's elements using Insertion sort

function insertionSort(arr){
    for(let i=1; i<arr.length; i++){
        // array with 1 element is always sorted
        let currVal = arr[i];

        let j;
        for(j=i-1; j>=0; j--){
            // check if previous value of array greater than current value 
            if(arr[j] > currVal){
                //then shift previous value to the unsorted region of array
                arr[j+1] = arr[j];
            }else{
                // arr[j] <= currVal
                break;
            }
        }
        // insert current value at its correct place
        arr[j+1] = currVal;
    }
};


// let arr = [2,3,4,5,6,7];
let arr = [7,2,6,1,5,3,4];

console.log("Unsorted Array: "+arr);
let t1 = performance.now();

insertionSort(arr);

let t2 = performance.now();
console.log("Sorted Array: "+arr);

console.log("Time taken: "+Math.floor(t2-t1));

File: 02.Array_problems/InsertionSort.java

// sort the array's elements using Insertion sort

package Array_problems;

import java.util.Arrays;

/**
 * InsertionSort
 */
public class InsertionSort {
    public static void insertionSort(int[] arr){
        for(int i=1; i<arr.length; i++){
            // array with 1 element is always sorted
            int currVal = arr[i];

            int j = 0;
            for(j=i-1; j>=0; j--){
                // check if the previous value of array is greater than current value
                if(arr[j] > currVal){
                    // then shift previous value to the unsorted region of array
                    arr[j+1] = arr[j];
                }else{
                    // arr[j] <= currVal
                    break;
                }
            }
            // insert current value 'currVal' to its correct position
            arr[j+1] = currVal;
        }
    }

    public static void main(String[] args) {
        int[] arr = {7,2,6,1,5,3,4};
        System.out.println("Unsorted Array: "+Arrays.toString(arr));
        insertionSort(arr);
        System.out.println("Sorted Array: "+Arrays.toString(arr));
    }
}